catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java

```diff
@@ -158,4 +158,85 @@ public void testAddPartition() {
+    String[] fieldDt1 = new String[] {columns[2].name()};
+    Literal<?> valueDt1 = Literals.dateLiteral(LocalDate.parse("2020-01-01"));
+    Partition partition1 =
+        Partitions.identity(
+            new String[][] {fieldCity1, fieldDt1}, new Literal<?>[] {valueCity1, valueDt1});
+    hiveTable.supportPartitions().addPartition(partition1);
+
+    // add partition: city=3/dt=2020-01-01
+    String partitionName2 = "city=3/dt=2020-01-01";
+    String[] fieldCity2 = new String[] {columns[1].name()};
+    Literal<?> valueCity2 = Literals.byteLiteral((byte) 3);
+    String[] fieldDt2 = new String[] {columns[2].name()};
+    Literal<?> valueDt2 = Literals.dateLiteral(LocalDate.parse("2020-01-01"));
+    Partition partition2 =
+        Partitions.identity(
+            new String[][] {fieldCity2, fieldDt2}, new Literal<?>[] {valueCity2, valueDt2});
+    hiveTable.supportPartitions().addPartition(partition2);
+
+    // add partition: city=3/dt=2020-01-02
+    String partitionName3 = "city=3/dt=2020-01-02";
+    String[] fieldCity3 = new String[] {columns[1].name()};
+    Literal<?> valueCity3 = Literals.byteLiteral((byte) 3);
+    String[] fieldDt3 = new String[] {columns[2].name()};
+    Literal<?> valueDt3 = Literals.dateLiteral(LocalDate.parse("2020-01-02"));
+    Partition partition3 =
+        Partitions.identity(
+            new String[][] {fieldCity3, fieldDt3}, new Literal<?>[] {valueCity3, valueDt3});
+    hiveTable.supportPartitions().addPartition(partition3);
+
+    // test drop one partition: city=2/dt=2020-01-01
+    hiveTable.supportPartitions().dropPartition(partitionName1, true);
+    NoSuchPartitionException exception1 =
+        Assertions.assertThrows(
+            NoSuchPartitionException.class,
+            () -> {
+              hiveTable.supportPartitions().getPartition(partitionName1);
+            });
+    Assertions.assertEquals(
+        String.format("Hive partition %s does not exist in Hive Metastore", partitionName1),
+        exception1.getMessage());
+
+    // test drop cascade partitions: city=3
+    hiveTable.supportPartitions().dropPartition(partitionName2, true);
+    NoSuchPartitionException exception2 =
+        Assertions.assertThrows(
+            NoSuchPartitionException.class,
+            () -> {
+              hiveTable.supportPartitions().getPartition(partitionName2);
+            });
+    Assertions.assertEquals(
+        String.format("Hive partition %s does not exist in Hive Metastore", partitionName2),
+        exception2.getMessage());
+    hiveTable.supportPartitions().dropPartition(partitionName3, true);
+    NoSuchPartitionException exception3 =
+        Assertions.assertThrows(
+            NoSuchPartitionException.class,
+            () -> {
+              hiveTable.supportPartitions().getPartition(partitionName3);
+            });
+    Assertions.assertEquals(
+        String.format("Hive partition %s does not exist in Hive Metastore", partitionName3),
```

Comment:
This test result is problematic. 

Why was partition `city=3/dt=2020-01-02` also deleted when you removed partition `city=3/dt=2020-01-01` above?